rustls/src/client/tls13.rs

```diff
@@ -63,102 +66,261 @@ static DISALLOWED_TLS13_EXTS: &[ExtensionType] = &[
     ExtensionType::ExtendedMasterSecret,
 ];
 
+/// `early_data_key_schedule` is `Some` if we sent the
+/// "early_data" extension to the server.
 pub(super) fn handle_server_hello(
     config: Arc<ClientConfig>,
     cx: &mut ClientContext<'_>,
     server_hello: &ServerHelloPayload,
-    mut resuming_session: Option<persist::Tls13ClientSessionValue>,
+    psks: Option<PresharedKeys>,
+    psk_modes: Vec<PSKKeyExchangeMode>,
     server_name: ServerName<'static>,
     mut randoms: ConnectionRandoms,
     suite: &'static Tls13CipherSuite,
     mut transcript: HandshakeHash,
-    early_key_schedule: Option<KeyScheduleEarly>,
+    early_data_key_schedule: Option<KeyScheduleEarly>,
     mut hello: ClientHelloDetails,
-    our_key_share: Box<dyn ActiveKeyExchange>,
+    our_key_share: Option<Box<dyn ActiveKeyExchange>>,
     mut sent_tls13_fake_ccs: bool,
     server_hello_msg: &Message<'_>,
     ech_state: Option<EchState>,
 ) -> hs::NextStateOrError<'static> {
+    // Early data requires a PSK.
+    //
+    // TODO(eric): Express this with types, like
+    // `../server/tls13::InputSecrets`.
```

Comment:
When do you plan to address these TODOs?

Reply:
I don't think they're critical for this feature,  so either I'll delete the comments or I can make those changes on a subsequent PR.